src/openms/source/KERNEL/MSExperiment.cpp

```diff
@@ -145,6 +149,306 @@ namespace OpenMS
     return ConstAreaIterator();
   }
 
+  MSExperiment::CoordinateType MSExperiment::aggregate(ConstAreaIterator begin, ConstAreaIterator end, AggregatorFunc rt_agg, AggregatorFunc mz_agg) const
+  {
+    /*std::vector<MSExperiment::CoordinateType> ity;
+    std::vector<MSExperiment::CoordinateType> tmp;
+    float rt = -1.0f;
+    for (auto it = begin; it != end; ++it)
+    {
+      if (it.getRT() != rt) 
+      {
+        rt = (float)it.getRT();
+        
+        ity.push_back(mz_agg(tmp));
+      }
+      else
+      {
+        tmp.push_back(it->getMZ());
+      }
+    }
+    // TODO we could allow SOA instead of AOS here
```

Comment:
Address the TODO comments in the code. If they are not relevant anymore, consider removing them. If they are, consider opening new issues to track them. [important]